.github/workflows/regression_check.py

```diff
@@ -0,0 +1,140 @@
+            )
+        github_comment += "\n"
+
+        return True, True, github_comment
+    return False, False, ""
+
+
+def check_changes_in_invalid_files() -> tuple[bool, bool, str]:
+    panics, possible_problems = check_file("result_not_valid_files.txt")
+
+    if len(panics) != 0 or len(possible_problems) != 0:
+        print(">>>>> Found changed non valid files (see above)")
+        for line in panics + possible_problems:
+            print(line)
+
+        github_comment = "\nFound differences in converting images that were not properly converted in develop branch.\n"
+        for line in panics:
+            github_comment += (
+                line.replace(RUST_PANIC, "") + " - panicked when checking\n"
+            )
+        for line in possible_problems:
+            github_comment += (
+                line.replace(POSSIBLE_PROBLEM, "") + " - changed visually \n"
+            )
+        github_comment += "\n"
+
+        return True, False, github_comment
+    return False, False, ""
+
+
+def check_changes_in_image_size() -> tuple[bool, bool, str]:
+    panics, possible_problems = check_file("result_image_size.txt")
+
+    if len(panics) != 0 or len(possible_problems) != 0:
+        print(">>>>> Found difference in size generated image(see above)")
+        for line in panics + possible_problems:
+            print(line)
+
+        github_comment = "\nGenerated png have different size in each run.\n"
+        for line in panics:
+            github_comment += (
+                line.replace(RUST_PANIC, "") + " - panicked when checking\n"
+            )
+        for line in possible_problems:
+            github_comment += (
+                line.replace(POSSIBLE_PROBLEM, "") + " - changed visually \n"
+            )
+        github_comment += "\n"
+
+        return True, True, github_comment
+    return False, False, ""
+
+
+def send_comment(github_comment: str) -> None:
+    with open(GITHUB_EVENT_PATH, "r") as file:
+        data = json.load(file)
+        comments_url = data["pull_request"]["comments_url"]
+
+    print(f"Sending comment to {comments_url}")
+
+    payload = json.dumps({"body": github_comment})
+    curl_command = f'curl -s -S -H "Authorization: token {GITHUB_TOKEN}" --header "Content-Type: application/vnd.github.VERSION.text+json" --data "{payload}" "{comments_url}"'
+    subprocess.run(curl_command, shell=True)
+
+
+if __name__ == "__main__":
+    if GITHUB_EVENT_NAME != "pull_request":
+        print("You are running this script in wrong event - expected pull_request, crashes may occur")
+
+    print(os.getcwd())
+
+    print()
+    create_comment_v, fail_ci_v, github_comment_v = check_changes_in_valid_files()
+    print()
+    create_comment_i, fail_ci_i, github_comment_i = check_changes_in_invalid_files()
+    print()
+    create_comment_s, fail_ci_s, github_comment_s = check_changes_in_image_size()
+    print()
+
+    create_comment = create_comment_v or create_comment_i or create_comment_s
```

Comment:
Is this checking whether `github_comment` created?
I guess `create_comment` is more like function name. The name `has_comment` would be cleaner :)